src/aggregate/functions/date.c

```diff
@@ -60,19 +60,20 @@ static int timeFormat(ExprEval *ctx, RSValue *result, RSValue **argv, size_t arg
 
 /* Fast alternative to mktime which is dog slow. From:
 https://gmbabar.wordpress.com/2010/12/01/mktime-slow-use-custom-function/ */
+// Fix and performance improvements:
+// https://godbolt.org/z/qscb5d9dT
+// https://quick-bench.com/q/oTV4_9uVqPTcrj2fpDEvbbMzZ48
+// https://quick-bench.com/q/2Bc8WY1Ys0vmbp-HPagWFxu81jI
 time_t fast_timegm(const struct tm *ltm) {
```

Comment:
pass-by-`const*` I presume because it's faster? How large is `struct tm`? has this been measured?

Reply:
56 bytes on my machine. I'm not sure what is your suggestion here

Reply:
My suggestion was to determine if it's faster to pass by `const*` than by value. But yes, 56 bytes is larger than what the Itanium ABI will pass by register (it will use only 6 registers before beginning to use the stack). So we gain nothing by passing by value in this case.

Reply:
We only use this function on local (stack) variables. I hope this function becomes in-line

Reply:
inlining is greatly aided by `static`. Which lets the compiler know that the function may never be called from outside this TU, and there does not need to be a symbol generated for it.

Reply:
I can add `static` here, although all the functions that uses it are static as well